android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -115,19 +120,36 @@ public void onCameraSwitchError(String s) {
      * Constructs a new {@code VideoCapturer} instance attempting to satisfy
      * specific constraints.
      *
-     * @param deviceId the ID of the requested video device. If not
-     * {@code null} and a {@code VideoCapturer} can be created for it, then
-     * {@code facingMode} is ignored.
+     * @param deviceId   the ID of the requested video device. If not
+     *                   {@code null} and a {@code VideoCapturer} can be created for it, then
+     *                   {@code facingMode} is ignored.
      * @param facingMode the facing of the requested video source such as
-     * {@code user} and {@code environment}. If {@code null}, "user" is
-     * presumed.
+     *                   {@code user} and {@code environment}. If {@code null}, "user" is
+     *                   presumed.
      * @return a {@code VideoCapturer} satisfying the {@code facingMode} or
      * {@code deviceId} constraint
      */
     private VideoCapturer createVideoCapturer(String deviceId, String facingMode) {
         String[] deviceNames = cameraEnumerator.getDeviceNames();
         List<String> failedDevices = new ArrayList<>();
 
+        // If deviceId is specified, then it takes precedence over facingMode.
+        if (deviceId != null) {
+            // if has uvc into id, instance uvc capturer
+            if (deviceId.contains("uvc")) {
+                String message = "Create UVC user-specified camera " + deviceId;
+                try {
+                    UvcCapturer videoCapturer = new UvcCapturer(deviceId, mUvcStrategy);
+                    Log.d(TAG, message + " succeeded");
+                    return videoCapturer;
+                } catch (Throwable tr) {
+                    failedDevices.add(deviceId);
```

Comment:
The fallback code won't work with this I think, because we are not enumerating the UVC cameras.

Reply:
Okay, so I'll remove the snippet.